vm/src/builtins/pystr.rs

```diff
@@ -767,6 +767,12 @@ impl PyStr {
         self.value.py_join(iter)
     }
 
+    // FIXME: two traversals of str is expensive
+    #[inline]
+    fn _to_char_idx(r: &str, byte_idx: usize) -> usize {
+        r.char_indices().take_while(|&(idx, _)| idx < byte_idx).count()
```

Comment:
```suggestion
        r[..byte_idx].chars().count()
```
is this equivalent code?

Reply:
Cool! Your suggestion looks much better!